packages/yarnpkg-core/sources/nodeUtils.ts

```diff
@@ -140,8 +140,7 @@ export function getCaller() {
 
 export function availableParallelism() {
   // TODO: Use os.availableParallelism directly when dropping support for Node.js < 19.4.0
-  if (`availableParallelism` in os)
-    // @ts-expect-error - No types yet
+  if (typeof os.availableParallelism !== `undefined`)
```

Comment:
Changed this check because otherwise `os` would have been `never` in the other branch.